Scripts/GodotSceneTranslationExtractor.cs

```diff
@@ -0,0 +1,210 @@
+                {
+                    HandleOptionsButton(match.Groups[1].Value, match.Groups[2].Value, path, lineNumber);
+                    continue;
+                }
+            }
+
+            match = GodotEditorDescription.Match(line);
+
+            if (match.Success)
+            {
+                HandleEditorDescription(match.Groups[1].Value);
+            }
+        }
+
+        // Output any remaining stuff if the last Godot node in the file had translations
+        foreach (var result in ProcessPreviousNode())
+        {
+            yield return result;
+        }
+    }
+
+    private void StartNode(string name, string? type)
+    {
+        translationsForCurrentNode.Clear();
+
+        lastNodeName = name;
+        lastNodeType = type;
+
+        ignoreLastNode = false;
+        ignoreLastNodeTooltip = false;
+    }
+
+    private void HandleProperty(string name, string value, string file, int line)
+    {
+        if (string.IsNullOrWhiteSpace(value))
+            return;
+
+        if (propertiesToLookFor.All(p => p != name))
+            return;
+
+        translationsForCurrentNode.Add((name, new ExtractedTranslation(value, file, line)));
+    }
+
+    private void HandleOptionsButton(string name, string data, string file, int line)
+    {
+        var values = data.Split(",", StringSplitOptions.TrimEntries);
+
+        if (values.Length % ItemListStride != 0)
+        {
+            ColourConsole.WriteErrorLine($"Failed to parse Godot items syntax at {file}:{line}");
+            return;
+        }
+
+        for (int i = 0; i < values.Length; i += ItemListStride)
+        {
+            var item = values[i];
+
+            if (item.StartsWith('\"'))
+                item = item.Substring(1, item.Length - 2);
+
+            translationsForCurrentNode.Add((name, new ExtractedTranslation(item, file, line)));
+        }
+    }
+
+    private IEnumerable<ExtractedTranslation> ProcessPreviousNode()
+    {
+        // Exit if no node is read yet
+        if (lastNodeName == null)
+            yield break;
+
+        if (ignoreLastNode && ignoreLastNodeTooltip)
+            yield break;
+
+        foreach (var (property, translation) in translationsForCurrentNode)
+        {
+            if (ignoreLastNode)
+            {
+                // Only tooltip allowed when a placeholder node
+                if (string.Compare(property, TooltipPropertyName, StringComparison.OrdinalIgnoreCase) == 0)
+                    yield return translation;
+
+                continue;
+            }
+
+            // The case where only tooltip would be ignored is not very useful, so we don't check that or support
+            // that combination of state variables here
+
+            // We are allowed to pass all translations through
+            yield return translation;
+        }
+    }
+
+    private void HandleEditorDescription(string description)
+    {
+        if (description.Contains(PlaceHolderMarker))
```

Comment:
`TOOLTIP_PLACEHOLDER` contains `PLACEHOLDER`, so ignoring tooltip translation will ignore the name as well?

By the way, *Placeholder* is used to make the UI in Godot editor look good. Tooltip isn't visible in Godot editor, so I guess there isn't a need to add placeholder tooltips?

Reply:
I just added the tooltip placeholder, just for feature completeness (i.e. if there ever is a case where a tooltip should already be ignored, it can be). Also in the logic I assumed that if tooltip is ignored the other parts are also ignored, so `TOOLTIP_PLACEHOLDER` containing `PLACEHOLDER` is intended.